enhancements/storage/storage-security-policy.md

```diff
@@ -0,0 +1,198 @@
+
+We want to replace admission plugin with Mutating Admission Policy once the feature becomes generally available. Once Mutating Admission Policy becomes GA, we propose to replace the admission hook with MAP on the lines of https://docs.google.com/document/d/13xa7imyvQzok3IJ66mheGqemQGhkNySl-He1h1iRuSM/edit?usp=sharing
+
+This would serve similar purpose as admission hook we are carrying as go code.
+
+### Proof of concept
+
+We have already implemented a proof-of-concept https://github.com/openshift/kubernetes/pull/2311/files, which works and does the job. But this uses existing SCC admission hook.
+
+
+### Workflow Description
+
+Admins that want to set defaults of `fsGroupChangePolicy` and `selinuxChangePolicy` will set `storage.openshift.io/fsgroup-change-policy` or `storage.openshift.io/selinux-change-policy`
+labels in corresponding namespaces.
+
+This will result in setting these fields in all pods in affected namespace if these fields are *not* already set.
+
+### API Extensions
+
+Adds following labels as API conventions for defaulting of security policies:
+
+- `storage.openshift.io/fsgroup-change-policy`
+- `storage.openshift.io/selinux-change-policy`
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+It should not have effect on Hypershift deployments.
+
+#### Standalone Clusters
+
+It should not have effect on Standalone deployments.
+
+#### Single-node Deployments or MicroShift
+
+It should not have effect on Single-Node deployments.
+
+### Implementation Details/Notes/Constraints
+
+We will implement the necessary admission hooks in - https://github.com/openshift/apiserver-library-go/tree/master/pkg ,
+which will be feature-gated while in Tech Preview.
+
+### Risks and Mitigations
+
+This feature requires storage to be enabled for this to work. See Operational Aspect section for more details.
+
+### Drawbacks
+
+The admission hook implementation should be replaced by Mutating Admission Hook and hence it is kinda busy work while we wait for MAP implementation to go GA in upstream.
+
+## Alternatives (Not Implemented)
+
+None
+
+## Open Questions [optional]
+
+None
+
+## Test Plan
+
+We will add e2e tests that validate if labels are working as intended. These would be either part of CSO tests or `openshift/origin` tests.
+
+## Graduation Criteria
+
+The feature will be behind a feature-gate during tech-preview.
+
+### Dev Preview -> Tech Preview
+
+N/A
+
+### Tech Preview -> GA
+
+During the initial development this feature will be behind a Openshift feature gate and will not be generally available. When we move this feature-gate to GA,
+it should be as simple as removal of the Openshift feature-gate.
+
+### Removing a deprecated feature
+
+N/A
+
+## Upgrade / Downgrade Strategy
+
+This should not have any impact on Upgrade and Downgrade strategy.
+
+## Version Skew Strategy
+
+Since this feature only applies to control-plane defautling, IMO we should be fine even older version of kubelets etc.
```

Comment:
The two fields that we set on podspec, when were they introduced to K8s?

Reply:
The fsgroupChangePolicy field has been GA for I dunno couple of years now (I am too lazy to dig this, since kube_feature.go does not even carry this field anymore).

`selinuxChangePolicy` went beta and enabled by default in v1.33

Reply:
Ok, so a 1.33 KAS is required to set the config?

During an upgrade, if there is a mix of 1.32 and 1.33 KAS, and this option gets set, is there any risk of getting the old 1.32 KAS panicking? Is it definitely safe to set this at the same point we are upgrading from 1.32 to 1.33? Is that possible with the current design?

Reply:
It should be safe even with a mix of 1.32 and 1.33 KAS, because KAS is not designed to panic and is coded to drop fields of an alpha features if feature is not enabled. I have tested and verified this multiple times.

Having said that - it does mean this option may get inconsistently applied during upgrade, but that doesn't seem like an end-of-the-world scenario. A feature that explicitly depends on specific version of KAS will always require specific version of KAS, I dunno if users will expect feature to work if mixed version of KAS is available.